src/digitalmarketplace/components/list-input/template.njk

```diff
@@ -0,0 +1,109 @@
+{% endif -%}
+
+{%- if params.errorMessage %}
+  {%- set listErrorId = id + '-error' -%}
+  {%- set describedBy = describedBy + ' ' + listErrorId if describedBy and listErrorId not in describedBy else listErrorId -%}
+{% endif -%}
+
+<div class="dm-list-input{% if params.classes %} params.classes {% endif %}{% if params.errorMessage %} govuk-form-group--error{% endif %}">
+  {% if params.errorMessage %}
+    {{ govukErrorMessage({
+      id: listErrorId,
+      classes: 'dm-list-input-error-message' + params.errorMessage.classes if params.errorMessage.classes,
+      attributes: params.errorMessage.attributes,
+      html: params.errorMessage.html,
+      text: params.errorMessage.text,
+      visuallyHiddenText: params.errorMessage.visuallyHiddenText
+    }) | indent(2) | trim }}
+  {% endif %}
+  {% call govukFieldset({
+    describedBy: params.fieldset.describedBy if params.fieldset.describedBy else describedBy,
+    classes: params.fieldset.classes,
+    legend: {
+      html: params.fieldset.legend.text + ('<span class="dm-list-input__optional__optional"> (Optional)</span>' if params.optional) ,
+      classes: 'dm-list-input__legend ' + ( params.fieldset.legend.classes if params.fieldset.legend.classes)
+    },
+    attributes: {
+      'data-module': 'dm-list-input'
+    }
+  }) %}
+
+  {%- if params.hint %}
+    {{ govukHint({
+        id: hintId,
+        classes: 'dm-list-input__hint ' + ( params.hint.classes if params.hint.classes),
+        attributes: params.hint.attributes,
+        html: params.hint.html,
+        text: params.hint.text
+      }) | indent(2) | trim() }}
+  {% endif -%}
+
+  <div class="dm-list-input__item-container" role="region" aria-live="assertive">
+    {% for i in range(0, params.maxItems) -%}
+      {%- set item = params.items[i] if params.items[i] else {} -%}
+      {%- set entryItemId = params.id + '-' + i -%}
+      {# Classes #}
+      {%- set itemClassesBase = 'govuk-input dm-list-input__item-input' -%}
+      {%- set itemClassesLarge = ' dm-list-input__item-input--large' if i > 99 else '' -%}
+      {%- set itemClassesExternal = ' ' + item.classes if item.classes else '' -%}
+      {%- set itemClassesError = ' govuk-input--error' if item.errorMessage else '' -%}
+      {%- if item.errorMessage %}
+        {% set errorId = entryItemId + '-error' %}
+      {% endif -%}
+
+      <div class="dm-list-input__item dm-list-input__item--visible {% if item.errorMessage %}dm-list-input__item--error govuk-form-group--error{% endif %}">
+        {% if item.errorMessage %}
+        {{ govukErrorMessage({
+          id: errorId,
+          classes: 'dm-list-input-error-message  ' + item.errorMessage.classes if item.errorMessage.classes,
+          attributes: item.errorMessage.attributes,
+          html: item.errorMessage.html,
+          text: item.errorMessage.text,
+          visuallyHiddenText: item.errorMessage.visuallyHiddenText
+        }) | indent(2) | trim }}
+        {% endif %}
```

Comment:
In the GOV.UK Design System error messages are after the question text and hint text [1](https://design-system.service.gov.uk/components/error-message/#how-it-works), is there a reason not to follow that design rule for this component?

Reply:
There's two places for error messages on this component:

I've moved the "component-level" error message as per that guidance (which I had missed - thank you!). This is for an error that affects the whole component (ie: "Enter at least one requirement").

The error message above is intended to be used at item level - tie an error to an input (ie: "too few characters", etc). Previously the `govukInput` it should attach to was hardcoded, but now that it's not, I'm going to tweak it to use the `errorMessage` property.